File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlValues.cs
using System.Text;

namespace T1.SqlSharp.Expressions;

public class SqlValues : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.Values;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_Values(this);
    }

    public List<ISqlExpression> Items { get; set; } = [];

    public string ToSql()
    {
        var sql = new StringBuilder();
        sql.Append("(");
        for (var i = 0; i < Items.Count; i++)
        {
            sql.Append(Items[i].ToSql());
            if (i < Items.Count - 1)
            {
                sql.Append(", ");
            }
        }
        sql.Append(")");
        return sql.ToString();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlValues.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression } from './ISqlExpression';

export class SqlValues implements ISqlExpression {
    SqlType: SqlType = SqlType.Values;
    Span: TextSpan = new TextSpan();
    Items: ISqlExpression[] = [];

    Accept(visitor: any): void {
        // 簡單實作
    }

    ToSql(): string {
        return `(${this.Items.map(item => item.ToSql()).join(', ')})`;
    }
} 